asistente_ladm_col/gui/queries/dockwidget_queries.py

```diff
@@ -28,214 +27,128 @@
                              QIcon, 
                              QCursor, 
                              QPixmap)
-from qgis.PyQt.QtNetwork import (QNetworkRequest, 
-                                 QNetworkAccessManager)
 from qgis.PyQt.QtWidgets import (QMenu, 
                                  QAction, 
                                  QApplication, 
                                  QLabel)
-from qgis.core import (Qgis,
-                       QgsFeature,
-                       QgsFeatureRequest,
-                       QgsExpression,
-                       QgsVectorLayer)
 from qgis.gui import (QgsDockWidget, 
                       QgsMapToolIdentifyFeature)
 
-from asistente_ladm_col.config.config_db_supported import ConfigDBsSupported
 from asistente_ladm_col.app_interface import AppInterface
 from asistente_ladm_col.lib.logger import Logger
-from asistente_ladm_col.config.layer_config import LayerConfig
-from asistente_ladm_col.config.general_config import (TEST_SERVER,
-                                                      SUFFIX_GET_THUMBNAIL)
+from asistente_ladm_col.config.general_config import SUFFIX_GET_THUMBNAIL
 
 from asistente_ladm_col.utils import get_ui_class
-from asistente_ladm_col.utils.utils import is_connected
-from asistente_ladm_col.utils.qt_utils import (ProcessWithStatus,
-                                               OverrideCursor)
+from asistente_ladm_col.utils.qt_utils import OverrideCursor
 
-from asistente_ladm_col.logic.ladm_col.tree_models import TreeModel
 
 DOCKWIDGET_UI = get_ui_class('dockwidgets/dockwidget_queries.ui')
 
 
 class DockWidgetQueries(QgsDockWidget, DOCKWIDGET_UI):
 
-    zoom_to_features_requested = pyqtSignal(QgsVectorLayer, list, dict, int)  # layer, ids, t_ids, duration
-
-    def __init__(self, iface, db, ladm_data, parent=None):
+    def __init__(self, iface, controller, parent=None):
         super(DockWidgetQueries, self).__init__(None)
         self.setupUi(self)
         self.setAllowedAreas(Qt.LeftDockWidgetArea | Qt.RightDockWidgetArea)
         self.iface = iface
-        self._db = db
-        self.ladm_data = ladm_data
-        self.logger = Logger()
+        self._controller = controller
 
+        self.logger = Logger()
         self.app = AppInterface()
```

Comment:
AppInterface is defined in the LADMQueryController, can we reuse it?

Reply:
Following your logic, AppInterface is also defined in AsistenteLADMCOLPlugin. Why don't we pass it then to DockWidgetQueries or to the controller itself? :thinking: 

It's a singleton.

Reply:
That's right, this is not a mistake. Just to keep it in mind we can say that the singleton classes are always instantiated in the class that we are going to use and we would not pass it as a parameter to any class.

Reply:
This is a singleton created to be used throughout the whole plugin. We get the instance on demand, we do it all the time.

Didn't get the point of the original question. Anyways, my answer to it was/is "nope, it's a singleton."